lib/config/guiconfiguration.go

```diff
@@ -166,13 +166,15 @@ func (c *GUIConfiguration) prepare() error {
 		c.APIKey = rand.String(32)
 	}
 
-	if c.WebauthnUserId == "" {
-		newUserId := make([]byte, 64)
+	if len(c.WebauthnUserId) == 0 {
+		// Spec recommends 64 random bytes, we use 32 and hex-encoding inflates to 64
+		newUserId := make([]byte, 32)
 		_, err := rand.Read(newUserId)
 		if err != nil {
 			return err
 		}
-		c.WebauthnUserId = base64.URLEncoding.EncodeToString(newUserId)
+		// Hex-encode the ID so that the raw binary values look nice in config.xml
```

Comment:
It's not really about "looking nice" only, but being printable and not conflict with XML syntax.

Reply:
Thanks, addressed in a2b4c30e827981a08ce000c96b6eff408478ed0d.